src/vm/hints/secp/ec_utils.rs

```diff
@@ -247,4 +400,335 @@ mod tests {
             )))
         );
     }
+
+    #[test]
+    fn run_compute_slope_ok() {
+        let hint_code = "from starkware.cairo.common.cairo_secp.secp_utils import SECP_P, pack\nfrom starkware.python.math_utils import line_slope\n\n# Compute the slope.\nx0 = pack(ids.point0.x, PRIME)\ny0 = pack(ids.point0.y, PRIME)\nx1 = pack(ids.point1.x, PRIME)\ny1 = pack(ids.point1.y, PRIME)\nvalue = slope = line_slope(point1=(x0, y0), point2=(x1, y1), p=SECP_P)";
+        let mut vm = VirtualMachine::new(
+            VM_PRIME.clone(),
+            vec![(
+                "range_check".to_string(),
+                Box::new(RangeCheckBuiltinRunner::new(true, bigint!(8), 8)),
+            )],
+            false,
+            &HINT_EXECUTOR,
+        );
+
+        //Insert ids.point0 and ids.point1 into memory
+        vm.memory = memory![
+            ((1, 0), 134),
+            ((1, 1), 5123),
+            ((1, 2), 140),
+            ((1, 3), 1232),
+            ((1, 4), 4652),
+            ((1, 5), 720),
+            ((1, 6), 156),
+            ((1, 7), 6545),
+            ((1, 8), 100010),
+            ((1, 9), 1123),
+            ((1, 10), 1325),
+            ((1, 11), 910)
+        ];
+
+        //Initialize fp
+        vm.run_context.fp = MaybeRelocatable::from((1, 14));
+
+        //Create ids
+        let ids = ids!["point0", "point1"];
+
+        //Create references
+        vm.references = HashMap::from([
+            (
+                0,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -14,
+                    offset2: 0,
+                    dereference: false,
+                    inner_dereference: false,
+                    immediate: None,
+                    ap_tracking_data: Some(ApTracking {
+                        group: 1,
+                        offset: 0,
+                    }),
+                },
+            ),
+            (
+                1,
+                HintReference {
+                    register: Register::FP,
+                    offset1: -8,
+                    offset2: 0,
+                    dereference: false,
+                    inner_dereference: false,
+                    immediate: None,
+                    ap_tracking_data: Some(ApTracking {
+                        group: 1,
+                        offset: 0,
+                    }),
+                },
+            ),
+        ]);
```

Comment:
And another one for references.

Reply:
Yes, I saw the references! macro, but in these cases where the offset1 is not auto-increment by 1, is not useful

```
for i in 0..$num {
    references.insert(i, HintReference::new_simple((i as i32 - $num)));
```
```
pub fn new_simple(offset1: i32) -> Self {
    HintReference {
        register: Register::FP,
        offset1,
        offset2: 0,
```

Reply:
Oh. We need a smarter one then.

Reply:
I will implement one in the following PR